cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -99,15 +96,12 @@ func enforceRequirements(flagSet *pflag.FlagSet, flags *applyPlanFlags, args []s
 		return nil, nil, nil, nil, err
 	}
 
-	initCfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, "upgrade/config", false, true)
+	initCfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, "upgrade/config", false, false)
```

Comment:
this is the main fix!

Reply:
Do we change the behavior introduced by #91980?

Reply:
Sorry, test did not cover this scenario.
Only if skipComponentConfigs is set to false, the componentConfig of kubelet and kubeproxy can be returned.

Reply:
> Do we change the behavior introduced by #91980?

this was already removed as a feature since its not possible to supply them with --config

Reply:
> Sorry, test did not cover this scenario. Only if skipComponentConfigs is set to false, the componentConfig of kubelet and kubeproxy can be returned.

yeah, it needs refractor for proper unit tests.

Reply:
/lgtm

Reply:
/lgtm